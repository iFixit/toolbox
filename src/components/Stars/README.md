### Examples

#### Default Stars
```
<Stars rating={3.4} onChange={value => alert(`Changed value to ${value}`)} />
```

#### Custom Color
```
<Stars
   rating={3.4}
   activeColor="#0071CE"
   onChange={value => alert(`Changed value to ${value}`)}
/>
```

#### More Than 5 Stars
```
<Stars
   rating={7.5}
   count={10}
   onChange={value => alert(`Changed value to ${value}`)}
/>
```
