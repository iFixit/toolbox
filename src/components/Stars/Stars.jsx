import React from 'react';
import PropTypes from 'prop-types';
import StarRatings from 'react-star-ratings';
import constants from '../../constants';

const { color } = constants;
const { string, number, func } = PropTypes;

const Stars = ({
   activeColor,
   count,
   inactiveColor,
   onChange,
   rating,
   size,
   spacing,
}) => (
   <StarRatings
      rating={rating}
      numberOfStars={count}
      changeRating={onChange && onChange}
      starDimension={`${size}px`}
      starEmptyColor={inactiveColor}
      starHoverColor={activeColor}
      starRatedColor={activeColor}
      starSpacing={`${spacing}px`}
   />
);

Stars.propTypes = {
   /** The rating. */
   rating: number.isRequired,
   /** The color of each active star. */
   activeColor: string,
   /** The number of total stars. */
   count: number,
   /** The color of each inactive star. */
   inactiveColor: string,
   /** Size in px of each star */
   size: number,
   /** Spacing in px between each star */
   spacing: number,
   /** Callback when a star is clicked. */
   onChange: func,
};

Stars.defaultProps = {
   activeColor: color.yellow[4],
   count: 5,
   inactiveColor: color.gray[3],
   size: 40,
   spacing: 0,
   onChange: null,
};

export default Stars;
