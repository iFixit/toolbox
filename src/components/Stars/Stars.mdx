---
name: Stars
route: /components/stars
menu: Components
---

import Stars from './Stars'
import { Playground, PropsTable } from 'docz'

# Stars

## Examples

### Default

<Playground>
  <Stars rating={2} onChange={value => alert(value)}/>
</Playground>

### More Than 5 Stars

<Playground>
  <Stars rating={7.5} count={10} onChange={value => alert(value)}/>
</Playground>

## Props

<PropsTable of={Stars} />
