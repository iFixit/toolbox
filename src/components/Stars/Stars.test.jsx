import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Stars from './Stars';

configure({ adapter: new Adapter() });

test('renders without crashing', () => {
   const tree = renderer.create(<Stars rating={3.5} />).toJSON();
   expect(tree).toMatchSnapshot();
});
