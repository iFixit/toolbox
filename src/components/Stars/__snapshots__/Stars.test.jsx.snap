// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="star-ratings"
  style={
    Object {
      "boxSizing": "border-box",
      "display": "inline-block",
      "position": "relative",
    }
  }
  title="3.5 Stars"
>
  <svg
    className="star-grad"
    style={
      Object {
        "height": "0",
        "position": "absolute",
        "visibility": "hidden",
        "width": "0",
        "zIndex": "0",
      }
    }
  >
    <defs>
      <linearGradient
        id={null}
        x1="0%"
        x2="100%"
        y1="0%"
        y2="0%"
      >
        <stop
          className="stop-color-first"
          offset="0%"
          style={
            Object {
              "stopColor": "#F9C310",
              "stopOpacity": "1",
            }
          }
        />
        <stop
          className="stop-color-first"
          offset="50%"
          style={
            Object {
              "stopColor": "#F9C310",
              "stopOpacity": "1",
            }
          }
        />
        <stop
          className="stop-color-final"
          offset="50%"
          style={
            Object {
              "stopColor": "#E1E1E2",
              "stopOpacity": "1",
            }
          }
        />
        <stop
          className="stop-color-final"
          offset="100%"
          style={
            Object {
              "stopColor": "#E1E1E2",
              "stopOpacity": "1",
            }
          }
        />
      </linearGradient>
    </defs>
  </svg>
  <div
    className="star-container"
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
    style={
      Object {
        "cursor": undefined,
        "display": "inline-block",
        "paddingLeft": undefined,
        "paddingRight": "0px",
        "position": "relative",
        "verticalAlign": "middle",
      }
    }
  >
    <svg
      className="widget-svg"
      style={
        Object {
          "height": "40px",
          "transform": undefined,
          "transition": "transform .2s ease-in-out",
          "width": "40px",
        }
      }
      viewBox="0 0 51 48"
    >
      <path
        className="star"
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        style={
          Object {
            "fill": "#F9C310",
            "transition": "fill .2s ease-in-out",
          }
        }
      />
    </svg>
  </div>
  <div
    className="star-container"
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
    style={
      Object {
        "cursor": undefined,
        "display": "inline-block",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "position": "relative",
        "verticalAlign": "middle",
      }
    }
  >
    <svg
      className="widget-svg"
      style={
        Object {
          "height": "40px",
          "transform": undefined,
          "transition": "transform .2s ease-in-out",
          "width": "40px",
        }
      }
      viewBox="0 0 51 48"
    >
      <path
        className="star"
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        style={
          Object {
            "fill": "#F9C310",
            "transition": "fill .2s ease-in-out",
          }
        }
      />
    </svg>
  </div>
  <div
    className="star-container"
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
    style={
      Object {
        "cursor": undefined,
        "display": "inline-block",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "position": "relative",
        "verticalAlign": "middle",
      }
    }
  >
    <svg
      className="widget-svg"
      style={
        Object {
          "height": "40px",
          "transform": undefined,
          "transition": "transform .2s ease-in-out",
          "width": "40px",
        }
      }
      viewBox="0 0 51 48"
    >
      <path
        className="star"
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        style={
          Object {
            "fill": "#F9C310",
            "transition": "fill .2s ease-in-out",
          }
        }
      />
    </svg>
  </div>
  <div
    className="star-container"
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
    style={
      Object {
        "cursor": undefined,
        "display": "inline-block",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "position": "relative",
        "verticalAlign": "middle",
      }
    }
  >
    <svg
      className="widget-svg multi-widget-selected"
      style={
        Object {
          "height": "40px",
          "transform": undefined,
          "transition": "transform .2s ease-in-out",
          "width": "40px",
        }
      }
      viewBox="0 0 51 48"
    >
      <path
        className="star"
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        style={
          Object {
            "fill": null,
            "transition": "fill .2s ease-in-out",
          }
        }
      />
    </svg>
  </div>
  <div
    className="star-container"
    onClick={null}
    onMouseEnter={null}
    onMouseLeave={null}
    style={
      Object {
        "cursor": undefined,
        "display": "inline-block",
        "paddingLeft": "0px",
        "paddingRight": undefined,
        "position": "relative",
        "verticalAlign": "middle",
      }
    }
  >
    <svg
      className="widget-svg"
      style={
        Object {
          "height": "40px",
          "transform": undefined,
          "transition": "transform .2s ease-in-out",
          "width": "40px",
        }
      }
      viewBox="0 0 51 48"
    >
      <path
        className="star"
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        style={
          Object {
            "fill": "#E1E1E2",
            "transition": "fill .2s ease-in-out",
          }
        }
      />
    </svg>
  </div>
</div>
`;
